scripts/scripts.js

```diff
@@ -47,6 +47,145 @@ export function createBlogDetails(data) {
+      const cardTags = card.getAttribute('data-tags').split(', ');
+      if (tags.length && !tags.some((tag) => cardTags.includes(tag))) {
+        card.style.display = 'none';
+      } else {
+        card.style.display = 'block';
+      }
+    });
+  };
+
+  // clear existing result cards
+  while (resultsContainer.firstChild) {
+    resultsContainer.removeChild(resultsContainer.firstChild);
+  }
+
+  // show result cards based on filtered data
+  data.forEach((result) => {
+    const resultCard = document.createElement('a');
+    const newPath = extractPath(result.url);
+    resultCard.href = `${window.location.origin}/${newPath}`;
+    resultCard.classList.add('result-card');
+    resultCard.setAttribute('data-tags', result.tags || '');
+    resultCard.innerHTML = `
+        <h2>${result.title}</h2>
+        <p>${result.snippet ? result.snippet : result.intro}</p>
+    `;
+    resultsContainer.appendChild(resultCard);
+  });
+
+  // clear existing tags
+  while (tagsPanel.firstChild) {
+    tagsPanel.removeChild(tagsPanel.firstChild);
+  }
+
+  const urlParams = new URLSearchParams(window.location.search);
+  const tagsFromUrl = urlParams.getAll('tags');
+  const selectedTags = tagsFromUrl.length ? tagsFromUrl : [];
+
+  //  render tags
+  const allTags = new Set(data.flatMap((result) => (result.tags || '').split(', ')));
+  allTags.forEach((tag) => {
+    const tagItem = document.createElement('div');
+    tagItem.innerHTML = `
+        <input type="checkbox" id="${tag}" value="${tag}">
+        <label for="${tag}">${tag} (0)</label>
+    `;
+    tagsPanel.appendChild(tagItem);
+  });
+
+  // add tag count number
+  const tagCounts = {};
+  data.forEach((result) => {
+    if (result.tags) {
+      const tags = result.tags.split(', ');
+      tags.forEach((tag) => {
+        if (tagCounts[tag]) {
+          tagCounts[tag] += 1;
+        } else {
+          tagCounts[tag] = 1;
+        }
+      });
+    }
+  });
+
+  // render tag counts
+  Object.keys(tagCounts).forEach((tag) => {
+    const label = tagsPanel.querySelector(`label[for="${tag}"]`);
+    label.textContent = `${tag} (${tagCounts[tag]})`;
+  });
+
+  // apply selected tag filter
+  filterResultsByTag(selectedTags);
+
+  // update url when tag checkbox is changed
+  tagsPanel.addEventListener('change', (e) => {
+    if (e.target.tagName === 'INPUT' && e.target.type === 'checkbox') {
+      // get all selected tags
+      const checkedBoxes = tagsPanel.querySelectorAll('input[type="checkbox"]:checked');
+      const currentlySelectedTags = Array.from(checkedBoxes)
+        .map((checkbox) => checkbox.value);
+
+      filterResultsByTag(currentlySelectedTags);
+
+      // update URL
+      const currentUrl = new URL(window.location.href);
+      currentUrl.searchParams.delete('tags');
+      currentlySelectedTags.forEach((tag) => {
+        currentUrl.searchParams.append('tags', tag);
+      });
+      window.history.pushState({}, '', currentUrl.toString());
```

Comment:
@Lauraa1003  This is just one part of the solution, what is still missing is: 

1. Read the tags search params on page load and check the corresponding checkboxes eg if you go to https://search--www--headwire-edge-delivery.hlx.page/search-results?q=seo&tags=Performance, the performance tag checkbox should be checked.
2. Similarly on history popstate, you need to read the search params and check the tag checkboxes if any. 

I propose to reduce the complexity by not handling the history at all (maybe do it later) but I would recommend to at least implement 1. so that if you share the search link with someone with checked tags, you get the same search results wdyt ?

Reply:
Yes agree, I'll change it!